src/Utility/Store.js

```diff
@@ -0,0 +1,311 @@
+/**
+ * Generic class that expose all data manipulation methods.
+ * A Store can be constructed with any Key => Value structure (Map like object).
+ * It exposes all methods similar to aarray methods in one place.
+ *
+ * @author KhaaZ
+ *
+ * @template T
+ * @class Store
+ * @prop {Map} cache - Key => Value Data structure
+ */
+class Store {
+    /**
+     * Creates an instance of Store.
+     *
+     * @param {Map} cache - Key => Value data structure
+     * @memberof Store
+     */
+    constructor(cache) {
+        this.cache = cache || new Map();
+    }
+
+    // Direct action on cache
+
+    /**
+     * Current size of the cache
+     *
+     * @readonly
+     * @type {Number}
+     * @memberof Store
+     */
+    get size() {
+        return this.cache.size;
+    }
+
+    /**
+     * Iterator over cache.entries.
+     *
+     * @returns {Iterator<Key, Value>}
+     * @memberof Store
+     */
+    entries() {
+        return this.cache.entries();
+    }
+
+    /**
+     * Iterator over cache.keys
+     *
+     * @returns {Iterator<Key>}
+     * @memberof Store
+     */
+    keys() {
+        return this.cache.keys();
+    }
+
+    /**
+     * Iterator over cache.values
+     *
+     * @returns {Iterator<Value>}
+     * @memberof Store
+     */
+    values() {
+        return this.cache.values();
+    }
+
+    /**
+     * Iterator over cache
+     *
+     * @returns {Iterator}
+     * @memberof Store
+     */
+    [Symbol.iterator]() {
+        return this.cache[Symbol.iterator]();
+    }
+
+    /**
+     * Wether an element exists in the Store
+     *
+     * @param {String} key
+     * @returns {Boolean}
+     * @memberof Store
+     */
+    has(key) {
+        return this.cache.has(key);
+    }
+
+    /**
+     * Get an element by key in the Store
+     *
+     * @param {String} key
+     * @returns {T}
+     * @memberof Store
+     */
+    get(key) {
+        return this.cache.get(key);
+    }
+
+    /**
+     * Set an element in the Store
+     *
+     * @param {String} key
+     * @param {T} value
+     * @returns {Store} - Current Store
+     * @memberof Store
+     */
+    set(key, value) {
+        this.cache.set(key, value);
+        return this;
+    }
+
+    /**
+     * Delete an element by key in the Store
+     *
+     * @param {String} key
+     * @returns {Boolean} - Whether the element was suppressed or not
+     * @memberof Store
+     */
+    delete(key) {
+        return this.cache.delete(key);
+    }
+
+    // General cache manipulation
+
+    /**
+     * Store.cache to array
+     * [ value, value, value ]
```

Comment:
```suggestion
     * `[ value, value, value ]`
```
Though idk why this is needed

Reply:
Probably not needed, I'll just remove it